src/core/Akka.Tests/Actor/AskSpec.cs

```diff
@@ -66,52 +66,63 @@ protected override void OnReceive(object message)
         }
 
         [Fact]
-        public void Can_Ask_actor()
+        public async Task Can_Ask_actor()
         {
             var actor = Sys.ActorOf<SomeActor>();
-            actor.Ask<string>("answer").Result.ShouldBe("answer");
+            var res = await actor.Ask<string>("answer");
+            res.ShouldBe("answer");
         }
 
         [Fact]
-        public void Can_Ask_actor_with_timeout()
+        public async Task Can_Ask_actor_with_timeout()
         {
             var actor = Sys.ActorOf<SomeActor>();
-            actor.Ask<string>("answer",TimeSpan.FromSeconds(10)).Result.ShouldBe("answer");
+            var res = await actor.Ask<string>("answer", TimeSpan.FromSeconds(10));
+            res.ShouldBe("answer");
         }
 
         [Fact]
-        public void Can_get_timeout_when_asking_actor()
+        public async Task Can_get_timeout_when_asking_actor()
         {
             var actor = Sys.ActorOf<SomeActor>();
-            Assert.Throws<AggregateException>(() => { actor.Ask<string>("timeout", TimeSpan.FromSeconds(3)).Wait(); });
```

Comment:
Here because we didn't use the async await syntax - we started to catch AggregateException and never actually checked the underneath type. And this probably caused us picking the wrong type of exception to throw(TaskCancelledException instead of AskTimeoutException)

Reply:
@Aaronontheweb Look at the code above. It's not actually "to be modern", it's just to help not to hide things.

Reply:
I see